pulsar-client/src/main/java/org/apache/pulsar/client/impl/TypedMessageBuilderImpl.java

```diff
@@ -284,7 +284,7 @@ public MessageMetadata getMetadataBuilder() {
 
     public Message<T> getMessage() {
         beforeSend();
-        return MessageImpl.create(msgMetadata, content, schema);
+        return MessageImpl.create(msgMetadata, content, schema, producer.topic);
```

Comment:
Why do we use `producer.topic` instead of `producer.getTopic()` ?

Reply:
@Demogorgon314  I would use producer.getTopic() in order to ease mocking.
also the producer field used to be `nullable` 
this change is the cause of https://github.com/streamnative/kop/issues/677
I am not sure that the KOP usage is correct, as it is using an internal API to build a message

if we change to producer.getTopic(), will you be able to fix the KOP issue ?
isn't there any other way of creating a Message without using the Builder ?

Reply:
If we change to `producer.getTopic()`, we are still unable fix the KoP issue, maybe we shouldn't use the `TypedMessageBuilderImpl` in KoP, we can construct `MessageImpl` manually or we can determine whether `producer ` is null or not  in `TypedMessageBuilderImpl#getMessage()`.